src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -21,11 +21,35 @@ namespace ray {
 namespace gcs {
 
 GcsResourceManager::GcsResourceManager(
+    instrumented_io_context &io_context,
     std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
     ClusterResourceManager &cluster_resource_manager)
-    : gcs_table_storage_(gcs_table_storage),
+    : io_context_(io_context),
+      gcs_table_storage_(gcs_table_storage),
       cluster_resource_manager_(cluster_resource_manager) {}
 
+void GcsResourceManager::Update(std::shared_ptr<const syncer::RaySyncMessage> message) {
```

Comment:
I found `Update` a bit confusing too... Is there a better name that could indicate this is from a syncer? Maybe like `SyncState`?

Reply:
Could you also share some reason why you think it's confusing?

Reply:
To me when I read this, it took some time to figure out this is a part of Reciever interface implementation. I feel like the function name is too general

Reply:
agreed. let's rename it to something very specific.